From b7b712912c8bf4004c3986d9c3f9913bf6bcb7e8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 07 Apr 2019 11:58:47 +0300
Subject: [PATCH] [SW-0] goya: pll: configure bypass

In bypass mode, we do not configure the DIV_SEL register assuming
the default is bypass.
Do not rely on the default value and always configure the DIV_SEL
register.

Change-Id: I8db8caa222263bae94227485e3da693e725c1b06
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/goya/pll.c b/goya/pll.c
index c30e1f5..a1ff3c2 100644
--- a/goya/pll.c
+++ b/goya/pll.c
@@ -164,12 +164,10 @@
 	void *reg;
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
-		if (cfg->div_factor[sel] == -1)
-			continue;
-
 		reg = cfg->base + HL_PLL_DIV_SEL + (sel * sizeof(u32));
 
-		if (bypass)
+		/* -1 means always bypass */
+		if (cfg->div_factor[sel] == -1 || bypass)
 			writel(HL_PLL_REF_CLK, reg);
 		else
 			writel(cfg->div_factor[sel] ?
